.github/workflows/libcxx-check-generated-files.yml

```diff
@@ -0,0 +1,29 @@
+name: "Check libc++ generated files"
+on:
+  pull_request_target:
+    paths:
+      - 'libcxx/**'
+permissions:
+  pull-requests: write
+
+jobs:
+  check_generated_files:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Fetch LLVM sources
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+
+      - name: Install clang-format
+        uses: aminya/setup-cpp@v1
+        with:
+          clangformat: 17.0.1
+
+      - name: Install Ninja
+        uses: seanmiddleditch/gha-setup-ninja@master
```

Comment:
Are we using these actions elsewhere? Because using random actions associated with the project might be a security risk?

Reply:
We could specify a different container and skip a lot of these steps, no?

Reply:
We have our own action for installing ninja:  llvm/actions/install-ninja

But do we really need this if the job is always going to run on Linux?  Can we just apt install it?

Reply:
Is it more idiomatic to use [llvm/actions/install-ninja](https://github.com/llvm/actions/tree/main/install-ninja) or to install it manually? I don't really mind either way. I think I've seen `uses:` being used more often in actions but I don't mind using `apt-get` directly either.

Reply:
If we build a container and run the job in that container, we'll save time each iteration of the job, no?

Reply:
Yes, I do want to investigate setting up a container for libc++ jobs and storing that in the GH Docker registry to gradually replace our current DockerHub thing, but I would like to tackle that as a separate task to improve the CI situation in the short term.